src/nanoemoji/write_font2svg.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Generate SVG files from the SVG font table."""
+from absl import app
+from absl import flags
+from absl import logging
+from fontTools import ttLib
+from lxml import etree
+from nanoemoji import codepoints
+from nanoemoji.color_glyph import map_viewbox_to_otsvg_space
+from nanoemoji import util
+import os
+import shutil
+import subprocess
+import sys
+from pathlib import Path
+from picosvg.svg import SVG
+from picosvg.svg_meta import strip_ns
+
+
+FLAGS = flags.FLAGS
+
+
+flags.DEFINE_string(
+    "output_dir", None, "Output dir. Files written to <glyph name>.svg."
+)
+flags.DEFINE_string(
+    "log_level",
+    "INFO",
+    "The threshold for what messages will be logged. One of DEBUG, INFO, WARN, "
+    "ERROR, or FATAL.",
+)
+
+
+def main(argv):
+    logging.set_verbosity(FLAGS.log_level)
+
+    font_file = util.only(lambda a: a.endswith(".ttf"), argv)
+    out_dir = Path(FLAGS.output_dir)
+    assert out_dir.is_dir(), f"{FLAGS.output_dir} is not a directory"
+
+    font = ttLib.TTFont(font_file)
+    assert "SVG " in font, f"No SVG table in {font_file}"
+    upem = font["head"].unitsPerEm
+    ascender = font["OS/2"].sTypoAscender
+    descender = font["OS/2"].sTypoDescender
+    metrics = font["hmtx"].metrics
+    logging.debug("Writing svgs from %s with to %s", font_file, out_dir)
+    logging.debug("upem %d ascender %d descender %d", upem, ascender, descender)
+
+    # We want a subsequent nanoemoji scale to be 1:1
+    # So use the font height (global) and width (per glyph) as the svg viewbox
+    height = ascender - descender
+
+    for raw_svg, min_gid, max_gid in font["SVG "].docList:
+        svg = SVG.fromstring(raw_svg)
+        svg_defs = etree.Element("defs")
+        svg_g = etree.Element("g")
+        svg_g.attrib["transform"] = f"translate(0, {ascender})"
+
+        for el in svg.svg_root:
+            if strip_ns(el.tag) == "defs":
+                svg_defs.append(el)
+            else:
+                svg_g.append(el)
+
+        svg.svg_root.append(svg_defs)
+        svg.svg_root.append(svg_g)
+
+        for gid in range(min_gid, max_gid + 1):
+            glyph_name = font.getGlyphName(gid)
+            width, _ = metrics[glyph_name]
+            svg.svg_root.attrib["viewBox"] = f"0 0 {width} {height}"
+            dest_file = out_dir / f"{glyph_name}.svg"
```

Comment:
there's a major problem using the glyph names as file names: case insensitive filesystems like the one on macOS or Windows. Both `Path("/tmp/A.svg")` and `Path("/tmp/a.svg")` point to the same file, so you end up overwriting one with the other. To work around this, if you want to continue using a file name that resembles the glyph name, you may need to use a glyph name to file name mapping scheme like UFO uses for encoding it's .glif file names (available under [`fontTools.ufoLib.filenames.userNameToFileName`](https://github.com/fonttools/fonttools/blob/ac4d5611ac793e4815f64eb05d46e0cb46ec4612/Lib/fontTools/ufoLib/filenames.py#L95)). So, e.g. "A" becomes "A_" and "a" stays "a", etc.

However since these svgs are only a temporary byproduct of the build, I would instead suggest that you avoid using the glyph name as the source for the file name and instead simply use the glyph index (integer, e.g. "a" becomes `glyph00032.svg` or whatever the gid), which is guaranteed to be unique (per font) and doesn't incur in the same case-insensitivity issue.

Reply:
Perhaps you didn't notice that because you chose a "unicase" test font like Bungee where A and a look the same :)

Reply:
What kind of monster would have a case insensitive filesystem?! [good point]

Reply:
welcome to non-linux

Reply:
I'm tempted to explore gid based naming. The glyph name is coming across via the glyphmap.